src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -4116,8 +4116,8 @@ public int Add(T item)
                 Debug.Assert(!this.TryGetValue(item, out i));
 #endif
                 int index = _firstRowId + _rows.Count;
-                this.AddItem(item, index);
                 _rows.Add(item);
+                this.AddItem(item, index);
```

Comment:
Just to be sure, we're only concerned about re-entrancy, not concurrency, correct?

Reply:
> Just to be sure, we're only concerned about re-entrancy, not concurrency, correct?

Yes, there is no concurrency in this component.